host/sgx/sgxload.c

```diff
@@ -180,36 +180,24 @@ static void* _allocate_enclave_memory(size_t enclave_size, int fd)
         int mflags = MAP_SHARED;
         uint64_t mmap_size = enclave_size;
 
-        /* If no file descriptor, then perform anonymous mapping and double
-         * the allocation size, so that BASE can be aligned on the SIZE
-         * boundary. This isn't neccessary on hardware backed enclaves, since
-         * the driver will do the alignment. */
-        if (fd == -1)
+        /* Perform an anonymous mapping and double the allocation size,
+         * so that BASE can be aligned on the SIZE boundary. This isn't
+         * necessary on hardware backed enclaves, since the driver will
+         * do the alignment. */
+        mflags |= MAP_ANONYMOUS | MAP_NORESERVE;
```

Comment:
Since this line is not inside any condition anymore, can we combine this line into line 180?

Reply:
Refactored into a single declaration per Anand's suggestion.